src/App.tsx

```diff
@@ -1,10 +1,76 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { Status } from './types/Status';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 11098;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [filterStatus, setFilterStatus] = useState(Status.ALL);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todosfromserver) => {
+        setTodos(todosfromserver);
+      })
+      .catch(() => {
+        setHasError(true);
+      });
+  }, []);
+
+  useEffect(() => {
+    if (hasError) {
+      const timeout = setTimeout(() => {
+        setHasError(false);
+      }, 3000);
+
+      return () => clearTimeout(timeout);
+    }
+
+    return () => { };
+  }, [hasError]);
+
+  const handleCheckboxChange = (todoId: number) => {
+    const updatedTodos = todos.map((todo) => {
+      if (todo.id === todoId) {
+        return { ...todo, completed: !todo.completed };
+      }
+
+      return todo;
+    });
+
+    setTodos(updatedTodos);
```

Comment:
It's better to use functional setState to ensure that the state update is based on the latest state value